components/dialog/dialog.js

```diff
@@ -0,0 +1,47 @@
+import '../colors/colors.js';
+import '../button/button-icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { DialogMixin } from './dialog-mixin.js';
+import { dialogStyles } from './dialog-styles.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { heading3Styles } from '../typography/styles.js';
+
+class Dialog extends DialogMixin(LitElement) {
```

Comment:
Curious about why things are split across the element and the mixin? Obviously the mixin would be handy to turn other things into dialogs, but we could defer that potentially.

Reply:
Yeah, that what my thinking was.  Much of the logic so far will be needed for `d2l-dialog-confirm`.  The logic in the mixin does require a certain amount of DOM though... I'd thought about moving that required DOM into a template helper in the mixin.

Reply:
Gotcha. The LitElement docs have an example that uses composed templates... so the mixin could just define stuff like `headerTemplate` and `footerTemplate` that the subclasses would use in their template. They don't have any examples I can find that use mixins actually.

Reply:
Yeah, I think I could perhaps move `render` to the mixin as the default, and break up to make it more useful if sub-classes want to define something slightly different.